homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump_hlrrwifi.py

```diff
@@ -0,0 +1,286 @@
+    def target_temperature(self) -> int | None:
+        """Return the temperature."""
+        if (
+            temperature := self.device.states[OverkizState.CORE_TARGET_TEMPERATURE]
+        ) and temperature.value_as_int:
+            return temperature.value_as_int
+
+        return None
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if (state := self.device.states[ROOM_TEMPERATURE_STATE]) and state.value_as_int:
+            return state.value_as_int
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self._global_control(target_temperature=int(temperature))
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if (state := self.device.states[LEAVE_HOME_STATE]) and state.value_as_str:
+            if state.value_as_str == OverkizCommandParam.ON:
+                return PRESET_HOLIDAY_MODE
+
+            if state.value_as_str == OverkizCommandParam.OFF:
+                return PRESET_NONE
+
+        return None
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if preset_mode == PRESET_HOLIDAY_MODE:
+            await self._global_control(leave_home=OverkizCommandParam.ON)
+
+        if preset_mode == PRESET_NONE:
+            await self._global_control(leave_home=OverkizCommandParam.OFF)
+
+    def _control_backfill(
+        self, value: str | None, state_name: str, fallback_value: str
+    ) -> str:
+        """Overkiz doesn't accept commands with undefined parameters. This function is guaranteed to return a `str` which is the provided `value` if set, or the current device state if set, or the provided `fallback_value` otherwise."""
+        if value:
+            return value
+        state = self.device.states[state_name]
+        if state and state.value_as_str:
+            return state.value_as_str
+        return fallback_value
+
+    async def _global_control(
+        self,
+        main_operation: str | None = None,
+        target_temperature: int | None = None,
+        fan_mode: str | None = None,
+        hvac_mode: str | None = None,
+        swing_mode: str | None = None,
+        leave_home: str | None = None,
+    ) -> None:
+        """Execute globalControl command with all parameters. There is no option to only set a single parameter, without passing all other values."""
+
+        main_operation = self._control_backfill(
+            main_operation, MAIN_OPERATION_STATE, OverkizCommandParam.ON
+        )
+        target_temperature = target_temperature or self.target_temperature
+        if not target_temperature:
+            raise homeassistant.exceptions.InvalidStateError(
+                "Target temperature not set"
+            )
```

Comment:
To understand, when can this happen? And how will this exception be handled. Should we not just set a default target temperature?

Reply:
I do not think that this can happen. The device always has a temperature setpoint, it is available in the device's state and we fallback to the current state in case it is missing from the command.

When the exception is raised, it is getting logged and a "toaster" notification appears in the UI (if the action was triggered from the UI). If we don't raise an exception and pass the command to the API with an undefined temperature, the behaviour will be the same but with a confusing Overkiz API error message instead.

I do not feel comfortable silently choosing and setting a default temperature on behalf of the user when something goes wrong internally.

Reply:
There is no known scenario where this would happen, so I have removed this test and exception.